htlcswitch/payment_result.go

```diff
@@ -46,3 +48,34 @@ type networkResult struct {
 	// which the failure reason might not be included.
 	isResolution bool
 }
+
+// serializeNetworkResult serializes the networkResult.
+func serializeNetworkResult(w io.Writer, n *networkResult) error {
+	if _, err := lnwire.WriteMessage(w, n.msg, 0); err != nil {
+		return err
+	}
+
+	return channeldb.WriteElements(w, n.unencrypted, n.isResolution)
+}
+
+// deserializeNetworkResult deserializes the networkResult.
+func deserializeNetworkResult(r io.Reader) (*networkResult, error) {
+	var (
+		err error
+	)
+
+	n := &networkResult{}
+
+	n.msg, err = lnwire.ReadMessage(r, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := channeldb.ReadElements(r,
```

Comment:
`r` on next line?

Reply:
This matches the style used in many other parts of the codebase